app/common/UserAPI.ts

```diff
@@ -1112,6 +1112,22 @@ export class DocAPIImpl extends BaseAPI implements DocAPI {
   }
 }
 
+interface DocWorkerInfo {
+  docWorkerUrl: string|null;
+  internalDocWorkerUrl: string|null;
+  selfPrefix?: string;
+}
```

Comment:
Perhaps a type like this would better reflect the expected states:
```suggestion
export type DocWorkerUrlInfo = {
  docWorkerUrl: string;
  internalDocWorkerUrl: string;
} | {
  selfPrefix: string;
};
```

Code that accepts such a type can distinguish between the cases in a type-safe way with a test like this:

```ts
  if ("docWorkerUrl" in docWorkerUrlInfo) {
    // docWorkerUrlInfo.docWorkerUrl and docWorkerUrlInfo.internalDocWorkerUrl
    // can safely be used
  } else {
    // docWorkerUrlInfo.selfPrefix can safely be used
  }
}
```

Reply:
Thanks! I will give it a try.

That would solve something I feel a bit odd: the absence of values are represented with both `null` for the URLs, and with `undefined` for the prefix. Feel a bit uncomfortable.

Anyway, I give it a try and in that case, that oddity would be solved for me :)

Reply:
Done!